scripts/cypress-replay.sh

```diff
@@ -0,0 +1,106 @@
+# exit when any command fails
+set -e
+
+##############################################################################
+# Metadata environment variables that can be overwritten by the build system #
+#                                                                            #
+# * NOTE: RECORD_REPLAY_API_KEY should already be set in the environment     #
+#                                                                            #
+##############################################################################
+
+# The command to run to run your tests (including starting servers if necessary)
+export TEST_COMMAND=${TEST_COMMAND:-'npm run test:e2e -- -- --browser "Replay Firefox"'}
+
+# The current commit SHA
+export RECORD_REPLAY_METADATA_SOURCE_COMMIT_ID=${RECORD_REPLAY_METADATA_SOURCE_COMMIT_ID:-$(git log -1 --pretty="format:%H")}
+
+##############################################################################
+# The build script with the following steps:                                 #
+#                                                                            #
+# * Run $TEST_COMMAND                                                        #
+# * Attach metadata to recorded tests                                        #
+# * Upload all tests                                                         #
+# * Generate a link to the test run on app.replay.io                         #
+#                                                                            #
+##############################################################################
+
+# keep track of the last executed command
+trap 'last_command=$current_command; current_command=$BASH_COMMAND' DEBUG
+# echo an error message before exiting
+trap 'EXIT_CODE=$? CMD=${last_command}; test $EXIT_CODE -ne 0 && echo "\n\n\"${CMD}\" command filed with exit code $EXIT_CODE."' EXIT
+
+# Set up Replay-specific environment variables
+export RECORD_REPLAY_TEST_RUN_ID=$(uuidgen | tr A-Z a-z)
+export RECORD_REPLAY_METADATA_FILE=$(mktemp)
+export RECORD_ALL_CONTENT=1
+export RECORD_REPLAY_API_SERVER=${RECORD_REPLAY_API_SERVER:-https://api.replay.io}
+
+CWD=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd );
+
+# Install dependencies for this script
+npm i --prefix $CWD @replayio/replay@0.0.0-experimental-747d6cb4 node-fetch@2.x
+
+# Clean up any prior recodings before starting
+npx @replayio/replay rm-all
```

Comment:
This, strictly speaking, might not be safe because this is global state, right? It's possible for multiple test runs to be running on the same CI box. We don't do this, but it's a supported buildkite agent.